packages/block-editor/src/components/block-list/block.js

```diff
@@ -664,6 +669,15 @@ function BlockListBlockProvider( props ) {
 		defaultClassName,
 	} = selectedProps;
 
+	// Users of the editor.BlockListBlock filter used to be able to
+	// access the block prop.
+	// Ideally these blocks would rely on the clientId prop only.
+	// This is kept for backward compatibility reasons.
+	const block = useMemo(
+		() => ( { ...blockWithoutAttributes, attributes } ),
+		[ blockWithoutAttributes, attributes ]
+	);
```

Comment:
The reason this is better than selector cache is because it's stored by component instance, not globally.